# Marko Melishchuk Notebook Big Ideas Final Project

In [10]:
!git clone https://github.com/oliviat10/Final-Project-MOK.git

Cloning into 'Final-Project-MOK'...
remote: Enumerating objects: 17, done.
remote: Counting objects: 100% (17/17), done.
remote: Compressing objects: 100% (14/14), done.
remote: Total 17 (delta 3), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (17/17), done.


Marko's Notebook

In [2]:
import numpy as np
import pandas as pd


air = pd.read_csv('toxicair.csv')


Imported my data file

In [3]:
air.head()

TRI_FACILITY_ID   DOC_CTRL_NUM       FACILITY_NAME      STREET_ADDRESS  \
0  15025SSCLR400ST  1396100016272  USS-CLAIRTON PLANT  400 STATE ST MS 71   
1  15025SSCLR400ST  1396100016272  USS-CLAIRTON PLANT  400 STATE ST MS 71   
2  15106THHMM235HO  1387010137127         O HOMMEL CO         235 HOPE ST   
3  15106THHMM235HO  1387010137127         O HOMMEL CO         235 HOPE ST   
4  15025SSCLR400ST  1392065615205  USS-CLAIRTON PLANT  400 STATE ST MS 71   

  CITY_NAME COUNTY_NAME  STATE_COUNTY_FIPS_CODE STATE_ABBR  ZIP_CODE  REGION  \
0  CLAIRTON   ALLEGHENY                   42003         PA     15025       3   
1  CLAIRTON   ALLEGHENY                   42003         PA     15025       3   
2  CARNEGIE   ALLEGHENY                   42003         PA     15106       3   
3  CARNEGIE   ALLEGHENY                   42003         PA     15106       3   
4  CLAIRTON   ALLEGHENY                   42003         PA     15025       3   

   ...  NAICS_CODES INDUSTRY_CODE    SRS_ID PFAS_IND ENVIRONMENTAL_MEDIUM  \
0  ...          NaN           331   14183.0        0              AIR FUG   
1  ...          NaN           331   14183.0        0            AIR STACK   
2  ...          NaN           325  149583.0        0              AIR FUG   
3  ...          NaN           325  149583.0        0            AIR STACK   
4  ...          NaN           331    5090.0        0              AIR FUG   

  RELEASE_RANGE_CODE  TOTAL_RELEASE  REL_EST_AMT  RELEASE_NA  \
0                NaN          380.0        380.0           0   
1                NaN            0.0          0.0           0   
2                NaN            NaN          0.0           0   
3                NaN         1784.0       1784.0           0   
4                NaN        16000.0      16000.0           0   

  RELEASE_BASIS_EST_CODE  
0                      O  
1                      O  
2                      O  
3                      O  
4                      E  

[5 rows x 38 columns]

Showing the beginning of the chart to give examples of what we are looking at

In [4]:
a_release = air['TOTAL_RELEASE'].sum()
print(a_release)

136976755.5773927


The line of code above sums the weight of all the total chemicals that have been released into Allegheny County Air over the past 30 years, which comes out to an astounding large number (in lbs)

In [5]:
air.shape

(25536, 38)

This describes the shape of the data so we know how descriptive the data set is

The total amount (in pounds) of the toxic chemical released to air, water, land, and underground injection wells during the calendar year (January 1 - December 31). This means that there are 136976755.5773927 lbs of toxic chemicals released into the air (so far).

Below: This line of code shows quite how many different toxic chemicals there are in the air by checking the names of the chemical compound listed in the category chemical

In [6]:
air['CHEM_NAME'].unique()

array(['Biphenyl', 'Lead', 'Ethylene', 'Naphthalene', 'Methanol',
       'Cresol (mixed isomers)', 'Ethyl acrylate', 'Toluene',
       'Cyanide compounds', 'Acetone', 'p-Cresol', 'Anthracene',
       'Barium compounds (except for barium sulfate (CAS No. 7727-43-7))',
       'Acrylic acid', 'Titanium tetrachloride', '1,2,4-Trimethylbenzene',
       'Phthalic anhydride', 'Maleic anhydride', 'Dibenzofuran',
       'Sodium hydroxide (solution)',
       'Nitrate compounds (water dissociable; reportable only when in aqueous solution)',
       'Ethylene glycol', 'Xylene (mixed isomers)',
       'Sulfuric acid (acid aerosols including mists, vapors, gas, fog, and other airborne forms of any particle size)',
       'Phosphoric acid', 'Zinc compounds',
       'Chromium compounds (except for chromite ore mined in the Transvaal Region)',
       'Nickel compounds', 'Certain glycol ethers', 'n-Butyl alcohol',
       'Butyl acrylate', 'p-Xylene', 'Ethylbenzene',
       'Molybdenum trioxide', 'Copper 

In [7]:
len(air['CHEM_NAME'].unique())

175

The above line counts how many different chemicals I just listed

There are 175 different chemicals or chemical compounds being released into the air currently

In [8]:
len(air['CITY_NAME'].unique())

49

In [9]:
air['TOTAL_RELEASE'].describe()

count    1.709100e+04
mean     8.014555e+03
std      6.910453e+04
min      0.000000e+00
25%      4.000000e+00
50%      1.222000e+02
75%      1.494000e+03
max      2.600800e+06
Name: TOTAL_RELEASE, dtype: float64

The above code shows some basic statistical analysis on the data and it counts how many instances of toxic chemical release there were

There are 17091 different releases of toxic gases (many come from multiple plants/facilities, it's not 17091 different facilities)

In [10]:
num_fac = len(air['FACILITY_NAME'].unique())

This line shows us how many different facilities there are that are releasing toxic chemicals into the air

In [11]:

print("There are 204 different facilities that release toxic gases")

There are 204 different facilities that release toxic gases


The line of code below cuts down the dataset to only the most important columns we need to determine the amount of toxic chemicals released into the air

In [12]:
a_done = air.drop(columns = ['TRI_FACILITY_ID', 'DOC_CTRL_NUM','STATE_ABBR','STATE_COUNTY_FIPS_CODE','NAICS_CODES','PFAS_IND','RELEASE_RANGE_CODE',
                    'ENVIRONMENTAL_MEDIUM','RELEASE_BASIS_EST_CODE','STREET_ADDRESS', 'REGION','RELEASE_NA','ENVIRONMENTAL_MEDIUM','SRS_ID',
                   'INDUSTRY_CODE','COUNTY_NAME','TOTAL_RELEASE','FACILITY_NAME','CITY_NAME','ASGN_FEDERAL_IND','ASGN_AGENCY',
                    'PARENT_CO_DB_NUM','STANDARDIZED_PARENT_COMPANY','EPA_REGISTRY_ID','TRADE_SECRET_IND','FAC_CLOSED_IND','PARENT_CO_NAME','REPORTING_YEAR',
                   'CAS_NUM','ELEMENTAL_METAL_INCLUDED','CHEM_NAME','LIST_3350','PRIMARY_SIC_CODE','SIC_CODES','PRIMARY_NAICS_CODE'])
a_done

ZIP_CODE CARCINOGEN CLEAN_AIR  REL_EST_AMT
0         15025          N         Y      380.000
1         15025          N         Y        0.000
2         15106          Y         Y        0.000
3         15106          Y         Y     1784.000
4         15025          N         N    16000.000
...         ...        ...       ...          ...
25531     15025          Y         Y        1.146
25532     15225          N         Y        1.000
25533     15225          N         Y        1.000
25534     15014          Y         Y      250.000
25535     15014          Y         Y     1600.000

[25536 rows x 4 columns]

Now this chart is narrowed down to four different catergories, each being: Zip code, Carcinogen (a substance that is defined as something that can cause cancer to humans), Clean air, and Relistic Estimated Amount. The way I will narrow down whether or a substance is dangerous is if the answer to the question of clean air is no (N in the chart) or is the answer to is a carcinogen is yes (Y in the carcinogen category). Then I will have a 2x2 chart which I will sum the total amount of toxic chemicals by zip code and rank zip codes by amount of toxic chemicals released.

In [13]:
air_dropped = air.drop(columns = ['CARCINOGEN','CLEAN_AIR','TRI_FACILITY_ID', 'DOC_CTRL_NUM','STATE_ABBR','STATE_COUNTY_FIPS_CODE','NAICS_CODES','PFAS_IND','RELEASE_RANGE_CODE',
                    'ENVIRONMENTAL_MEDIUM','RELEASE_BASIS_EST_CODE','STREET_ADDRESS', 'REGION','RELEASE_NA','ENVIRONMENTAL_MEDIUM','SRS_ID',
                   'INDUSTRY_CODE','COUNTY_NAME','TOTAL_RELEASE','FACILITY_NAME','CITY_NAME','ASGN_FEDERAL_IND','ASGN_AGENCY',
                    'PARENT_CO_DB_NUM','STANDARDIZED_PARENT_COMPANY','EPA_REGISTRY_ID','TRADE_SECRET_IND','FAC_CLOSED_IND','PARENT_CO_NAME','REPORTING_YEAR',
                   'CAS_NUM','ELEMENTAL_METAL_INCLUDED','CHEM_NAME','LIST_3350','PRIMARY_SIC_CODE','SIC_CODES','PRIMARY_NAICS_CODE'])

After filtering this all out I created multiple query masks to filter out the facilites that aren't releasing dangerous chemicals into the air.
To do this I create 3 groups that I will filter by: are there carcinogens released (they do not have to be in the air for the release to be marked as yes, but if the air is deemed clean then the release of chemicals isn't going into the air, it's just going somewhere else) is the air determined to be clean, and if there is a release of toxic chemicals into the air is it a large amount

This code checks if the release is a carcinogen

In [54]:
query_mask = a_done['CARCINOGEN'] != 'N'
query_mask.head(5)

0    False
1    False
2     True
3     True
4    False
Name: CARCINOGEN, dtype: bool

Example

In [55]:
car_no = a_done[query_mask]
car_no.head(5)

ZIP_CODE CARCINOGEN CLEAN_AIR  REL_EST_AMT
2      15106          Y         Y          0.0
3      15106          Y         Y       1784.0
6      15225          Y         Y        250.0
7      15225          Y         Y        250.0
12     15144          Y         Y          0.0

In [57]:
qquery_mask = a_done['CLEAN_AIR'] == 'N'
qquery_mask.head()

0    False
1    False
2    False
3    False
4     True
Name: CLEAN_AIR, dtype: bool

In [72]:
qqquery_mask = a_done['REL_EST_AMT'] >= 100
qqquery_mask.head()

0     True
1    False
2    False
3     True
4     True
Name: REL_EST_AMT, dtype: bool

In [73]:
final = a_done[(query_mask & qquery_mask) & qqquery_mask]
final.head(500)

ZIP_CODE CARCINOGEN CLEAN_AIR  REL_EST_AMT
4893      15025          Y         N       1300.0
7482      15025          Y         N        230.0
7483      15025          Y         N       1613.0
8442      15025          Y         N        185.0
8443      15025          Y         N       1085.0
8942      15144          Y         N        250.0
8943      15144          Y         N        250.0
18812     15144          Y         N        250.0
20735     15025          Y         N       1430.0